plugins/woocommerce/includes/wc-conditional-functions.php

```diff
@@ -136,6 +136,20 @@ function is_checkout_pay_page() {
 	}
 }
 
+if ( ! function_exists( 'is_pay_for_order_page' ) ) {
+
+	/**
+	 * is_pay_for_order_page - Returns true when viewing the pay for order page.
+	 *
+	 * @param bool $check_key Optional. If true, check if the key is set in the URL.
+	 * @return bool
+	 */
+	function is_pay_for_order_page( $check_key = false ) {
```

Comment:
It feels like this function should be more specific to WooCommerce, either by calling is `wc_is_pay_for_order_page()` or `is_wc_pay_for_order_page()`.

Reply:
Hmm, not sure since all the other helpers follow a similar pattern: is_checkout_pay_page, is_checkout, is_cart, is_product, etc.

Reply:
(but I renamed `is_admin_settings_page` in https://github.com/woocommerce/woocommerce/pull/59069/commits/a275a1afaddeb02afde5468bdb7f2f1d21e13388)